### ISSUE TITLE
Accessibility evaluation of playlist pages

### ISSUE THREAD

**Issue by elynema** [avalonmediasystem/avalon#6312]
# Description

SiteImprove cannot scan playlists (unless it somehow gets ahold of a public one). Since these are user-facing pages, we should do a proactive usability evaluation.

# Done Looks Like

- [ ] Check with SiteImprove browser plugin
 - [ ] Playlist page (lists all playlists)
 - [ ] Create playlist page
 - [ ] Edit playlist page
 - [ ] View playlist page
- [ ] Check all pages with at least one other accessibility tool: ANDI bookmarklet or WAVE tool
- [ ] navigate through all pages with keyboard only
- [ ] Create issues for any identified problems with WCAG 2.1 AA

**Comment by Dananji** [comment]
Playlist index page:
| Success criterion | WCAG 2.1 Level | Accessibility checker | Error | Relevant HTML element |
| --- | --- | --- | --- | --- |
| 4.1.2 Name, Role, Value | Level A | SiteImprove & WAVE| Form field missing a label | Tag filter drop-down menu |
| 1.4.3 Contrast (Minimum) | Level AA | SiteImprove & WAVE | Color contrast (2.85:1) doesn't meet min requirement | Disabled `Previous` button on the table pagination |
| 1.4.3 Contrast (Minimum) | Level AA | SiteImprove & WAVE | Color contrast (2.54:1) doesn't min requirement | Highlighted page number button on the table pagination |
| 1.4.3 Contrast (Minimum) | Level AA | WAVE | Color contrast (2.84:1) doesn't meet min requirement | Avalon release version text on the footer of the page (this is not playlist specific) |

Keyboard accessibility:
- Cannot reach buttons in delete popover with keyboard navigation

Create playlist page:
| Success criterion | WCAG 2.1 Level | Accessibility checker | Error | Relevant HTML element |
| --- | --- | --- | --- | --- |
| 4.1.2 Name, Role, Value | Level A | SiteImprove & WAVE | Form field missing a label | `<input>` textbox for `Name` |
| 4.1.2 Name, Role, Value | Level A | SiteImprove & WAVE | Form field missing a label | `<span>` combobox for `Tags` |
| 4.1.2 Name, Role, Value | Level A | SiteImprove & WAVE | Form field missing a label | `<input>` textbox for `Tags` |

Playlist edit page:
| Success criterion | WCAG 2.1 Level | Accessibility checker | Error | Relevant HTML element |
| --- | --- | --- | --- | --- |
| 4.1.2 Name, Role, Value | Level A | SiteImprove & WAVE| Button missing a text alternative | Playlist item edit button with SVG |
| 1.4.3 Contrast | Level AA | SiteImprove & WAVE | Color contrast (3.95:1) doesn't meet min requirement | `No description` and `No tags` text in in playlist details |
| 4.1.2 Name, Role, Value | Level A | SiteImprove & WAVE | Form field missing a label | `<input>` form field for playlist item position | 
| 4.1.2 Name, Role, Value | Level A | SiteImprove & WAVE | Form field missing a label | `<input>` textbox for `Name` |
| 4.1.2 Name, Role, Value | Level A | SiteImprove & WAVE | Form field missing a label | `<span>` combobox for `Tags` |
| 3.3.2 Labels or Instructions | Level A | WAVE & axeDevtools | Ensure an element's role supports its ARIA attributes | Edit playlist details button and cancel edits button |

Keyboard accessibility:
- Edit playlist details button (button with SVG icon next to `Playlist Details`) is not in the tab order, thus making it not available for a keyboard user.
- Cancel playlist details edit button is not in the tab order, thus making it not reachable via keyboard.

View playlist page:
| Success criterion | WCAG 2.1 Level | Accessibility checker | Error | Relevant HTML element |
| --- | --- | --- | --- | --- |
| 2.5.3 Label in Name | Level A | SiteImprove | Visible label and accessible name do not match | Playlist item links in `StructuredNavigation` component in Ramp |
| 4.1.2 Name, Role, Value | Level A | SiteImprove & WAVE  | Form field missing a label | `<input>` textbox for both `Name` and `Time` in edit marker form | 
| 3.3.2 Labels or Instructions | Level A | SiteImprove & axeDevtool | ARIA attribute unsupported or prohibited | `aria-selected` on anchor tags for Share panel |

Keyboard accessibility:
- Playlist items container does not have visible focus when using tab navigation. When it is reached using tab navigation there is not visual indicator that the focus is on playlist items container after `Share` button.
- Cannot reach the accordion style panels for `Markers` and `Source Item Details` using keyboard navigation.
